operation/manager.go

```diff
@@ -139,21 +151,10 @@ func (sm *SingleOperationManager) WaitForSuccess(
 	}
 }
 
-func (sm *SingleOperationManager) cancelInLock(ctx context.Context) {
-	myOp := ctx.Value(somCtxKeySingleOp)
-	op := sm.currentOp
-
-	if op == nil || myOp == op {
-		return
-	}
-
-	op.cancelFunc()
-
-	sm.currentOp = nil
```

Comment:
Because canceling now blocks, we can trust the operation itself to manage the `sm.currentOp` state completely.